src/components/core/Markdown/Bold.tsx

```diff
@@ -0,0 +1,87 @@
+import IconContainer from './IconContainer';
+import { useCallback } from 'react';
+
+export default function Bold({
+  selectedRange,
+  textAreaRef,
+  setSelectedRange,
+  setUserDragged,
+}: {
+  selectedRange: number[];
+  textAreaRef: React.RefObject<HTMLTextAreaElement>;
+  setSelectedRange: (value: number[] | ((prevVar: number[]) => number[])) => void;
+  setUserDragged: (value: boolean) => void;
+}) {
+  const handleClick = useCallback(() => {
+    setUserDragged(false);
+
+    const textArea = textAreaRef.current;
+    if (textArea) {
+      const [start, end] = selectedRange;
+      const selectedText = textArea.value.substring(start, end);
+      const selectedTextWithSurrounding = textArea.value.substring(start - 2, end + 2);
+
+      const selectedTextIsBold = selectedText.startsWith('**') && selectedText.endsWith('**');
+      const selectedTextIsSurroundedBold =
+        selectedTextWithSurrounding.startsWith('**') && selectedTextWithSurrounding.endsWith('**');
+
+      // If the selected text includes any bold tags, remove them
+      if (selectedTextIsBold) {
+        const newText = textArea.value.replace(selectedText, selectedText.slice(2, -2));
```

Comment:
it looks like this'll replace the first occurrence of `selectedText`, so if there's more than one instance of the same bold text, it may cause the 1st one to be bolded when you meant to unbold the 2nd

Reply:
for example if the text was 

`**this is bold**

**this is bold**`

and you highlighted the second `**this is bold**` and click the bold button

Reply:
an alternative could be to use the start/end indices to replace the exact area of the text instead